indico/modules/events/registration/client/js/form/fields/DateInput.jsx

```diff
@@ -28,7 +28,8 @@ function DateInputComponent({value, onChange, disabled, required, dateFormat, ti
     const timeString = timeFormat ? timeValue : '00:00:00';
     onChange(`${dateString}T${timeString}`);
   };
-  const handleTimeChange = newTime => onChange(`${dateValue}T${newTime.format('HH:mm:00')}`);
+  const handleTimeChange = newTime =>
+    onChange(`${dateValue}T${newTime ? newTime.format('HH:mm:00') : ''}`);
```

Comment:
```suggestion
    onChange(newTime ? `${dateValue}T${newTime.format('HH:mm:00')}` : dateValue);
```

Untested, but I think that's more readable..